changelog/7126.bugfix.rst

```diff
@@ -0,0 +1,3 @@
+Switched to ``saferepr`` to format bytes ``parametrize`` parameters
+for ``--setup-show`` output to prevent errors when Python is called with ``-bb``
+to catch bytearray with unicode comparison.
```

Comment:
This line is not accurate in this case -- `-bb` errors not due to comparison but because of `str(b'abc')`.

Overall I don't think the cause is very important, only the fact that it failed. So I'd omit the "to catch ..." part.